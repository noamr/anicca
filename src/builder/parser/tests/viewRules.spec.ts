/// </ reference "jest" />
import nearley from 'nearley'
const grammar = require('../viewRules.ne')

const parse = (str: string) => new nearley.Parser(grammar).feed(str).finish()[0]

describe('ViewRules', () => {
    it('rules', () => {
        expect(parse('on click')).toMatchSnapshot()
        expect(parse('on keydown when event.code==12')).toMatchSnapshot()
        expect(parse('attribute src')).toMatchSnapshot()
        expect(parse('content')).toMatchSnapshot()
        expect(parse('data custom')).toMatchSnapshot()
        expect(parse('style border-width')).toMatchSnapshot()
        expect(parse('style --yoyo')).toMatchSnapshot()
        expect(parse('id')).toMatchSnapshot()
        expect(parse('class')).toMatchSnapshot()
        expect(parse('for [id, value] in filter(children, value() > 1)')).toMatchSnapshot()
    })
})
