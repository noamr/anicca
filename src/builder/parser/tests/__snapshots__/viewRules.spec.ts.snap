// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewRules rules 1`] = `
Object {
  "eventType": "click",
  "type": "DomEvent",
}
`;

exports[`ViewRules rules 2`] = `
Object {
  "attribute": "src",
  "type": "BindAttribute",
}
`;

exports[`ViewRules rules 3`] = `
Object {
  "type": "BindContent",
}
`;

exports[`ViewRules rules 4`] = `
Object {
  "attribute": "custom",
  "type": "BindData",
}
`;

exports[`ViewRules rules 5`] = `
Object {
  "style": "border-width",
  "type": "BindStyle",
}
`;

exports[`ViewRules rules 6`] = `
Object {
  "style": "--yoyo",
  "type": "BindStyle",
}
`;
